tests/integration/trace/test_zipkin_trace.py

```diff
@@ -236,3 +198,64 @@ def test_span_host_field():
     assert thrift_obj.spanHost.ipv4 == ipv4_to_int(host_span.ipv4)
     assert thrift_obj.spanHost.port == host_span.port
     assert thrift_obj.spanHost.serviceName == host_span.service_name
+
+
+@pytest.mark.gen_test
+def test_no_infinite_trace_submit():
+
+    def submit(request):
+        count[0] += 1
+        # three time (1 time as client, 2 times as server)
+        if count[0] == 3:
```

Comment:
General feedback: the assertion below is about your test's behavior and not the code you're trying to test. If we still had an infinite loop (submitting millions of traces) this wouldn't catch it because it would stop submitting traces after it's seen 3. It's better to let the code submit all the traces it needs to and then assert that only happened 3 times.